File: 605-can-place-flowers/can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        start,end=0,1
        res=0
        lfb=len(flowerbed)
        
        if lfb==1 and flowerbed[0]==0: return True

        while(end<lfb):
            if res==n: return True
            win= flowerbed[start:end+1]
            if sum(win)==0: 
                res+=1
                start=end
            else: start+=1
            
            end=start+2
        
        if start+2==lfb:
            win = flowerbed[start:]
            if sum(win)==0: res+=1
     

        return res>=n

File: 605-can-place-flowers/can-place-flowers.ex
defmodule Solution do
  @spec can_place_flowers(flowerbed :: [integer], n :: integer) :: boolean
  def can_place_flowers(flowerbed, n) do
    chunks = Enum.chunk_every([0] ++ flowerbed ++ [0], 3, 1, :discard)
    solve(chunks, n)
  end

  defp solve(_, 0), do: true

  defp solve([], _), do: false

  defp solve([_], n) when n > 1, do: false

  defp solve([[0,0,0]], 1), do: true

  defp solve([[0, 0, 0], h2 | rest], n) do
    solve([[1 | tl(h2)] | rest], n - 1)
  end

  defp solve([_ | rest], n), do: solve(rest, n)
end